ethers-solc/src/compile/output/mod.rs

```diff
@@ -471,3 +594,23 @@ impl<'a> fmt::Display for OutputDiagnostics<'a> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_link_order() {
+        let mut dependency_tree = DependencyTree::new();
+        dependency_tree.insert("Contract", vec!["Library", "ThirdLibrary"]);
+        dependency_tree.insert("Library", vec!["SecondLibrary"]);
+        dependency_tree.insert("SecondContract", vec!["Library"]);
+        dependency_tree.insert("ThirdContract", vec!["ThirdLibrary", "FourthLibrary"]);
+        dependency_tree.insert("FourthLibrary", vec!["Library"]);
+        dependency_tree.insert("SecondLibrary", vec![]);
+        dependency_tree.insert("ThirdLibrary", vec![]);
+
+        let link_order = get_link_order(&dependency_tree);
+        assert_eq!(link_order, vec!["SecondLibrary", "Library", "ThirdLibrary"]);
```

Comment:
would be better if this was `Vec<(target, lib)>` so we can just iterate over those pairs, rn we iterate over all artifacts in the project in `link_all_dynamic` even if they dont need to be linked